### PR TITLE
restructure directory, add jsdoc(documentation), refactor code

### PR DESCRIPTION
**Pull request by kryptokinght**
1. Removed small bugs :beetle: : 
2. Restructure the whole directory into a neat folder structure. :beer: 
3. Moved the functions inside main.js into separate files, added them into a separate lib folder. :file_folder: 
4. main.js is now less then 100 lines. :smile: 
3. Added documentation :pen: 
4. Updated README.md to help others know how to add documentation :scroll: 
5. Removed .gitignore bug for data.json :bug:

### CODE REVIEW

File: lib/checkHistory.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+const path = require('path');
+
+const PATH_TO_DATA = path.resolve(__dirname, "../data.json");
```
Comment (thealphadollar) [inline lib/checkHistory.js]: Let's store this file in user's home directory.

File: lib/genWish.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+function genWish() {
```
Comment (thealphadollar) [inline lib/genWish.js]: make this function accept an argument (first name of the person), and accordingly modify the birthday wishes to include first name. It is more personalized.

Comment (kryptokinght) [reply lib/genWish.js]: @thealphadollar made necessary changes.

**Comment by thealphadollar** [comment]
LGTM.

**Pull request merged**